rotkehlchen/chain/evm/decoding/airdrops.py

```diff
@@ -0,0 +1,78 @@
+from abc import ABCMeta
+from typing import TYPE_CHECKING, Optional
+
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.decoding.cowswap.constants import CPT_COWSWAP
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.accounting.structures.evm_event import EvmEvent
+    from rotkehlchen.assets.asset import Asset, EvmToken
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.fval import FVal
+    from rotkehlchen.types import ChecksumEvmAddress
+    from rotkehlchen.user_messages import MessagesAggregator
+
+CLAIMED = b'\xd46\xe9\x97=\x1eD\xd4\r\xb4\xd4\x11\x9e<w<\xad\xb12;&9\x81\x96\x8c\x14\xd3\xd1\x91\xc0\xe1H'  # noqa: E501
+
+
+def make_claim(
+        event: 'EvmEvent',
+        user_address: 'ChecksumEvmAddress',
+        amount: 'FVal',
+        asset: 'Asset',
+        counterparty: str,
+        notes: Optional[str] = None,
+) -> bool:
+    """It matches a transfer event to an airdrop claim, changes the required fields
+     then returns `True` if a match was found"""
+    if not (event.event_type == HistoryEventType.RECEIVE and event.location_label == user_address and amount == event.balance.amount and asset == event.asset):  # noqa: E501
+        return False
+
+    event.event_subtype = HistoryEventSubType.AIRDROP
+    event.counterparty = counterparty
+    event.notes = f'Claim {amount} {asset.symbol_or_name()} from {counterparty} airdrop' if notes is None else notes  # noqa: E501
+    return True
+
+
+class CommonAirdropsDecoder(DecoderInterface, metaclass=ABCMeta):
```

Comment:
This has nothing in common with the rest. It's specific to cowswap

```suggestion
class CowswapAirdropDecoder(DecoderInterface, metaclass=ABCMeta):
```